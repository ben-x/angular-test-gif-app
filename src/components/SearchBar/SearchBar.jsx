import React from 'react';
import {SearchBarDiv, SearchDiv, SearchBox, SearchButton} from './SearchBar.styles';


const SearchBar = ({handleChange, onSubmit}) =>{

  return (
    <SearchBarDiv>
      <div />
    <SearchDiv>
      <SearchBox placeholder='search here' onChange={handleChange} autoFocus/>
      <SearchButton onClick={onSubmit}>
      <i class="fa fa-search" aria-hidden="true"></i></SearchButton>
    </SearchDiv>
      <div />
    </SearchBarDiv>
  )
}

export default SearchBar;
