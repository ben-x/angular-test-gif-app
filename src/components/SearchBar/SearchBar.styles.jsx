import styled from 'styled-components';

const SearchDiv = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: none;
  margin: 5% 0px;
`;

const SearchBox = styled.input`
  border-radius:15px;
  padding: 10px;
  border: solid 2px #f5f7f6;
`;

export {SearchDiv, SearchBox}
