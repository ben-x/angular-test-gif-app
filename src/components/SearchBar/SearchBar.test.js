import React from "react";
import { shallow } from "enzyme";
import SearchBar from "./SearchBar";

const props = {
    handleChange: jest.fn(),
    onSubmit: jest.fn()
};

const wrap = shallow(<SearchBar {...props} />);

describe("<SearchBar />", () => {
    it("renders the component successfully", () => {
        expect(wrap).toMatchSnapshot();
    });
});
