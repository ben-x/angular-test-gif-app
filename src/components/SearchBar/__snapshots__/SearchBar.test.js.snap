// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> renders the component successfully 1`] = `ShallowWrapper {}`;
